Adding HttpClient examples to send the certificate (cert_auth.md)

damienbod opened the pull request:
Added HttpClient examples, showing different ways to send the certificate in a HttpClient request
Fixes #15974

damienbod commented:
https://github.com/aspnet/AspNetCore.Docs/issues/15974

guardrex commented:
btw @damienbod ... If the PR "fixes" the issue, use `Fixes #{ISSUE NUMBER}` so that when the PR merges the issue will automatically close.

Rick-Anderson commented:
@damienbod  like this
Fixes #15974

etortec commented on aspnetcore/security/authentication/certauth.md:
````diff
@@ -311,7 +311,85 @@ namespace AspNetCoreCertificateAuthApi
+}
+```
+
+#### Implement an HttpClient using a certificate and the X-ARR-ClientCert HTTP Header
````
I'd suggest to totally remove the now obsolete `Implement an HttpClient using a certificate and the X-ARR-ClientCert HTTP Header`, or at least to add a warning that the example code in that section can solely be used for API testing purposes and is not how you should implement the API client (as explained in #15974).

damienbod replied:
@etortec @blowdart I removed this and updated the forwarding handling for Azure as this is an Azure example

damienbod commented:
Fixes #16621

damienbod commented:
@blowdart Are we good here now? 

I removed the forwarding for Azure. And added a section for custom web proxies which uses certificate forwarding like before. Thanks for your help.

blowdart commented:
Yea, as long as you've called out the middleware still needs to be present. Just doesn't have to be configured.

damienbod commented:
@blowdart Added the note

damienbod commented:
Hi @blowdart Would it be possible to review this?

Thanks Damien

damienbod commented:
@Rick-Anderson @scottaddie 

Anything we need to do to get this completed?

Greetings Damien

jim-brown-fst commented:
> Yea, as long as you've called out the middleware still needs to be present. Just doesn't have to be configured.

When you say the middleware needs to be present, does that mean this won't work with Azure Functions? I can't add the middleware, but is there a workaround I can use without the middleware to get Azure to forward my certificate in the header?
